tests/bench/explicit.py

```
@@ -0,0 +1,1184 @@
+    
+    print(f"\nEvaluating {len(all_traces)} trace(s)...\n")
+
+    # Process traces in batches if batch_size > 1
+    results: list[EvaluationResult] = []
+
+    async def process_trace(idx: int, trace: dict[str, Any], source_file: str) -> EvaluationResult | None:
+        """Process a single trace and return the result."""
+        try:
+            props = extract_propositions(trace)
+            if not props:
+                logger.warning(f"Trace {idx} from {source_file} has no propositions, skipping")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)